### ISSUE TITLE
dependency problem for shutdown.target in the user instance

### ISSUE THREAD

**Issue by aroig** [systemd/systemd#4322]
### Submission type
- [x] Bug report
- [ ] Request for enhancement (RFE)

_NOTE: Do not submit anything other than bug reports or RFEs via the issue tracker!_
### systemd version the issue has been seen with

> 231

_NOTE: Do not submit bug reports about anything but the two most recently released systemd versions upstream!_
### Used distribution

> arch linux
### In case of bug report: Expected behaviour you didn't see

> Starting a service that depends on shutdown.target in the user instance, should stop all units with default dependencies.
### In case of bug report: Unexpected behaviour you saw

> Starting a service that depends on shutdown.target in the user instance results in: 
> 
> Failed to start test.service: Transaction contains conflicting jobs 'stop' and 'start' for shutdown.target. 
> Probably contradicting requirement dependencies configured.
> 
> shutdown.target is not started, and no unit gets stopped.
### In case of bug report: Steps to reproduce the problem

test.service:

```
[Unit]
DefaultDependencies=no
Requires=shutdown.target

[Service]
Type=oneshot
ExecStart=/usr/bin/true
```

Starting it on the --user instance:

```
$ systemctl --user start test.service
Failed to start test.service: Transaction contains conflicting jobs 'stop' and 'start' for shutdown.target. Probably contradicting requirement dependencies configured. 
See user logs and 'systemctl --user status test.service' for details.
```

It turns out `test.service` implicitly depends on `-.slice`, despite `DefaultDependencies=no`

```
$ systemctl --user show test.service | grep Requires            
Requires=shutdown.target -.slice
RequiresMountsFor=/home/abdo
```

And `-.slice` conflicts with `shutdown.target`, causing the conflicting job.

```
$ systemctl --user show -- -.slice | grep Conflicts
Conflicts=shutdown.target
```

This always happens when exiting a session, because of `systemd-exit.service`, then `shutdown.target` is not started and the services are not stopped cleanly.

```
$ systemctl --user show systemd-exit.service | grep Requires
Requires=shutdown.target -.slice
RequiresMountsFor=/home/abdo
```

**Comment by poettering** [comment]
I figure we need to install "-.slice" as unit file for the --user instance too.

**Comment by poettering** [comment]
Fix waiting in #4326. Please have a look

**Comment by aroig** [comment]
I'm confused now.

I built #4326 to test it on a vm, but it turns out that on the vm, with systemd 231 and arch linux, there is no problem to begin with: `-.slice` does not conflict with `shutdown.target`.

On my laptop, on the other hand, `-.slice` conflicts with `shutdown.target`. Same distro and same systemd version as in the vm. Do you have any idea what is going on? Who creates `-.slice`?

**Comment by poettering** [comment]
If there's no unit file for "-.slice" then one is synthesized.

Is it possible that your laptpo runs something older than 417800228f69db9ff37f4193d11508757dd09308 and friends?

Can you provide the "systemctl show -- -.slice" output for base systems, plus the systemd package versions for both?

**Comment by aroig** [comment]
Both laptop and qemu vm are on arch linux with systemd 231-4

Laptop:

```
$ systemctl --user show -- -.slice
ControlGroup=/user.slice/user-1000.slice/user@1000.service
MemoryCurrent=18446744073709551615
CPUUsageNSec=18446744073709551615
TasksCurrent=249
Delegate=no
CPUAccounting=no
CPUShares=18446744073709551615
StartupCPUShares=18446744073709551615
CPUQuotaPerSecUSec=infinity
IOAccounting=no
IOWeight=18446744073709551615
StartupIOWeight=18446744073709551615
BlockIOAccounting=no
BlockIOWeight=18446744073709551615
StartupBlockIOWeight=18446744073709551615
MemoryAccounting=no
MemoryLow=0
MemoryHigh=18446744073709551615
MemoryMax=18446744073709551615
MemoryLimit=18446744073709551615
DevicePolicy=auto
TasksAccounting=yes
TasksMax=18446744073709551615
Id=-.slice
Names=-.slice
RequiredBy=xrandr-single.service flag.slice sys-kernel-debug.mount dpms-off.service dbus.service apps.slice daemons.slice weston.socket stream-stop.service rtags.socket mpd.socket tint2.service vncserver.service xrandr-dual.service tmp.mount pulseaudio.socket sys-fs-fuse-connections.mount wm.slice setenv.service adb.socket xkeymap.service run-user-1000-gvfs.mount dev-mqueue.mount xrdb.service worker.slice machine.slice mpc-pause.service xrandr-mirror.service systemd-reboot.service dev-hugepages.mount systemd-poweroff.service boot.mount systemd-wmquit.service init.scope systemd-suspend.service dpms-on.service audio.slice workers.slice chat-away.service dbus.socket plank.service emacs.socket sys-kernel-config.mount gpg-agent-update.service i3lock.service
Conflicts=shutdown.target
Before=systemd-poweroff.service dpms-on.service xrandr-mirror.service apps.slice workers.slice machine.slice setenv.service init.scope weston.socket wm.slice dpms-off.service i3lock.service gpg-agent-update.service pulseaudio.socket run-user-1000-gvfs.mount systemd-suspend.service chat-away.service flag.slice shutdown.target emacs.socket dbus.socket xrandr-dual.service dbus.service stream-stop.service tint2.service daemons.slice xrandr-single.service xrdb.service systemd-wmquit.service audio.slice boot.mount vncserver.service mpd.socket adb.socket worker.slice dev-mqueue.mount mpc-pause.service plank.service sys-kernel-config.mount systemd-reboot.service dev-hugepages.mount rtags.socket tmp.mount sys-kernel-debug.mount sys-fs-fuse-connections.mount xkeymap.service
Documentation=man:systemd.special(7)
Description=Root Slice
LoadState=loaded
ActiveState=active
SubState=active
StateChangeTimestamp=Mon 2016-10-10 12:32:41 CEST
StateChangeTimestampMonotonic=55048029
InactiveExitTimestamp=Mon 2016-10-10 12:32:41 CEST
InactiveExitTimestampMonotonic=55048029
ActiveEnterTimestamp=Mon 2016-10-10 12:32:41 CEST
ActiveEnterTimestampMonotonic=55048029
ActiveExitTimestampMonotonic=0
InactiveEnterTimestampMonotonic=0
CanStart=no
CanStop=no
CanReload=no
CanIsolate=no
StopWhenUnneeded=no
RefuseManualStart=yes
RefuseManualStop=yes
AllowIsolate=no
DefaultDependencies=no
OnFailureJobMode=replace
IgnoreOnIsolate=yes
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=no
AssertResult=no
ConditionTimestampMonotonic=0
AssertTimestampMonotonic=0
Transient=no
StartLimitIntervalSec=10000000
StartLimitBurst=5
StartLimitAction=none
```

VM:

```
$ systemctl --user show -- -.slice
ControlGroup=/user.slice/user-1000.slice/user@1000.service
MemoryCurrent=18446744073709551615
CPUUsageNSec=18446744073709551615
TasksCurrent=18446744073709551615
Delegate=no
CPUAccounting=no
CPUShares=18446744073709551615
StartupCPUShares=18446744073709551615
CPUQuotaPerSecUSec=infinity
IOAccounting=no
IOWeight=18446744073709551615
StartupIOWeight=18446744073709551615
BlockIOAccounting=no
BlockIOWeight=18446744073709551615
StartupBlockIOWeight=18446744073709551615
MemoryAccounting=no
MemoryLow=0
MemoryHigh=18446744073709551615
MemoryMax=18446744073709551615
MemoryLimit=18446744073709551615
DevicePolicy=auto
TasksAccounting=yes
TasksMax=18446744073709551615
Id=-.slice
Names=-.slice
RequiredBy=adb.socket gpg-agent-update.service dropdown.slice mpc-pause.service weston.socket dev-mqueue.mount vncserver.service xrandr-single.service plank.service xrandr-mirror.service apps.slice audio.slice xkeymap.service dbus.socket stream-stop.service i3lock.service init.scope yoga\x5cx2drotate.slice rtags.socket setenv.service dpms-on.service pulseaudio.socket -.mount dev-hugepages.mount tmp.mount emacs.socket xrdb.service sys-kernel-config.mount yoga\x5cx2dtablet.slice xrandr-dual.service run-user-1000.mount dpms-timeout.service wm.slice dpms-off.service tint2.service chat-away.service daemons.slice sys-kernel-debug.mount
Before=vncserver.service xrandr-single.service dpms-on.service xkeymap.service init.scope apps.slice dev-hugepages.mount gpg-agent-update.service dev-mqueue.mount dpms-timeout.service stream-stop.service tmp.mount sys-kernel-debug.mount xrdb.service -.mount audio.slice mpc-pause.service chat-away.service i3lock.service yoga\x5cx2dtablet.slice tint2.service daemons.slice xrandr-dual.service dropdown.slice setenv.service dpms-off.service pulseaudio.socket emacs.socket rtags.socket yoga\x5cx2drotate.slice plank.service xrandr-mirror.service dbus.socket adb.socket run-user-1000.mount wm.slice weston.socket sys-kernel-config.mount
Documentation=man:systemd.special(7)
Description=Root Slice
LoadState=loaded
ActiveState=active
SubState=active
StateChangeTimestamp=Mon 2016-10-10 15:55:07 CEST
StateChangeTimestampMonotonic=81858640
InactiveExitTimestamp=Mon 2016-10-10 15:55:07 CEST
InactiveExitTimestampMonotonic=81858640
ActiveEnterTimestamp=Mon 2016-10-10 15:55:07 CEST
ActiveEnterTimestampMonotonic=81858640
ActiveExitTimestampMonotonic=0
InactiveEnterTimestampMonotonic=0
CanStart=no
CanStop=no
CanReload=no
CanIsolate=no
StopWhenUnneeded=no
RefuseManualStart=yes
RefuseManualStop=yes
AllowIsolate=no
DefaultDependencies=no
OnFailureJobMode=replace
IgnoreOnIsolate=yes
NeedDaemonReload=no
JobTimeoutUSec=infinity
JobTimeoutAction=none
ConditionResult=no
AssertResult=no
ConditionTimestampMonotonic=0
AssertTimestampMonotonic=0
Transient=no
StartLimitIntervalSec=10000000
StartLimitBurst=5
StartLimitAction=none
```

**Comment by poettering** [comment]
Hmm, I see no real difference except for the shutdown.target thing. Could you diff -u the two outputs?

**Comment by aroig** [comment]
```
$ diff -u laptop.txt vm.txt
--- laptop.txt  2016-10-10 16:18:51.477888643 +0200
+++ vm.txt  2016-10-10 16:18:55.851304481 +0200
@@ -1,7 +1,7 @@
     ControlGroup=/user.slice/user-1000.slice/user@1000.service
     MemoryCurrent=18446744073709551615
     CPUUsageNSec=18446744073709551615
-    TasksCurrent=249
+    TasksCurrent=18446744073709551615
     Delegate=no
     CPUAccounting=no
     CPUShares=18446744073709551615
@@ -23,20 +23,19 @@
     TasksMax=18446744073709551615
     Id=-.slice
     Names=-.slice
-    RequiredBy=xrandr-single.service flag.slice sys-kernel-debug.mount dpms-off.service dbus.service apps.slice daemons.slice weston.socket stream-stop.service rtags.socket mpd.socket tint2.service vncserver.service xrandr-dual.service tmp.mount pulseaudio.socket sys-fs-fuse-connections.mount wm.slice setenv.service adb.socket xkeymap.service run-user-1000-gvfs.mount dev-mqueue.mount xrdb.service worker.slice machine.slice mpc-pause.service xrandr-mirror.service systemd-reboot.service dev-hugepages.mount systemd-poweroff.service boot.mount systemd-wmquit.service init.scope systemd-suspend.service dpms-on.service audio.slice workers.slice chat-away.service dbus.socket plank.service emacs.socket sys-kernel-config.mount gpg-agent-update.service i3lock.service
-    Conflicts=shutdown.target
-    Before=systemd-poweroff.service dpms-on.service xrandr-mirror.service apps.slice workers.slice machine.slice setenv.service init.scope weston.socket wm.slice dpms-off.service i3lock.service gpg-agent-update.service pulseaudio.socket run-user-1000-gvfs.mount systemd-suspend.service chat-away.service flag.slice shutdown.target emacs.socket dbus.socket xrandr-dual.service dbus.service stream-stop.service tint2.service daemons.slice xrandr-single.service xrdb.service systemd-wmquit.service audio.slice boot.mount vncserver.service mpd.socket adb.socket worker.slice dev-mqueue.mount mpc-pause.service plank.service sys-kernel-config.mount systemd-reboot.service dev-hugepages.mount rtags.socket tmp.mount sys-kernel-debug.mount sys-fs-fuse-connections.mount xkeymap.service
+    RequiredBy=adb.socket gpg-agent-update.service dropdown.slice mpc-pause.service weston.socket dev-mqueue.mount vncserver.service xrandr-single.service plank.service xrandr-mirror.service apps.slice audio.slice xkeymap.service dbus.socket stream-stop.service i3lock.service init.scope yoga\x5cx2drotate.slice rtags.socket setenv.service dpms-on.service pulseaudio.socket -.mount dev-hugepages.mount tmp.mount emacs.socket xrdb.service sys-kernel-config.mount yoga\x5cx2dtablet.slice xrandr-dual.service run-user-1000.mount dpms-timeout.service wm.slice dpms-off.service tint2.service chat-away.service daemons.slice sys-kernel-debug.mount
+    Before=vncserver.service xrandr-single.service dpms-on.service xkeymap.service init.scope apps.slice dev-hugepages.mount gpg-agent-update.service dev-mqueue.mount dpms-timeout.service stream-stop.service tmp.mount sys-kernel-debug.mount xrdb.service -.mount audio.slice mpc-pause.service chat-away.service i3lock.service yoga\x5cx2dtablet.slice tint2.service daemons.slice xrandr-dual.service dropdown.slice setenv.service dpms-off.service pulseaudio.socket emacs.socket rtags.socket yoga\x5cx2drotate.slice plank.service xrandr-mirror.service dbus.socket adb.socket run-user-1000.mount wm.slice weston.socket sys-kernel-config.mount
     Documentation=man:systemd.special(7)
     Description=Root Slice
     LoadState=loaded
     ActiveState=active
     SubState=active
-    StateChangeTimestamp=Mon 2016-10-10 12:32:41 CEST
-    StateChangeTimestampMonotonic=55048029
-    InactiveExitTimestamp=Mon 2016-10-10 12:32:41 CEST
-    InactiveExitTimestampMonotonic=55048029
-    ActiveEnterTimestamp=Mon 2016-10-10 12:32:41 CEST
-    ActiveEnterTimestampMonotonic=55048029
+    StateChangeTimestamp=Mon 2016-10-10 15:55:07 CEST
+    StateChangeTimestampMonotonic=81858640
+    InactiveExitTimestamp=Mon 2016-10-10 15:55:07 CEST
+    InactiveExitTimestampMonotonic=81858640
+    ActiveEnterTimestamp=Mon 2016-10-10 15:55:07 CEST
+    ActiveEnterTimestampMonotonic=81858640
     ActiveExitTimestampMonotonic=0
     InactiveEnterTimestampMonotonic=0
     CanStart=no
```

**Comment by poettering** [comment]
Hmm, no idea where that extra might be coming from I must admit... Maybe from the other side?

Can you do the same systemctl show diff dance for shutdown.target?

**Comment by aroig** [comment]
Here it is. Sorry, there is a ton of crap in there... The user units on both are similar, but the ones at the vm are older. I'm going to try to boot the vm and my laptop with minimal user units, to see what happens.

```
$ diff -u laptop.txt vm.txt
--- laptop.txt  2016-10-10 16:36:29.217904652 +0200
+++ vm.txt  2016-10-10 16:37:18.948807151 +0200
@@ -1,7 +1,8 @@
 Id=shutdown.target
 Names=shutdown.target
-ConflictedBy=homedir-gc@lib.service clean-backups.timer openbox.service compton.service termite@10.service chat-away.service xrdb.service weston.socket vncserver.service sockets.target homedir-gc@music.timer plank.service homedir-gc@audio.service gpg-agent.service flag.slice jwatch.service journal-dropdown.service homedir-check@lib.service stream-stop.service homedir-gc@bbrepos.service udev-wifi-sta.target homedir-check@lib.timer timers.target audio.slice clean-trash.service termite@9.service network-online.target homedir-gc@vids.timer adb-proxy.service emacs-save-quit.service homedir-gc@music.service vm-disk-snapshot@virtdata.service pulseaudio.socket graphical.target clean-tmp.service xrandr-single.service laptop.target clean-build.service housekeeping.target awesome.service adb.socket systemd-networkd-wait-online@bd\x5cx2dnet.service at-spi-bus.service termite-dropdown.service rtags.socket termite@5.service wmclean.target tint2.service vm-disk-snapshot@arch.service dbus.socket chromium@0.service homedir-gc@pics.service termite@2.service termite@8.service gconf.service emacs.service workers.slice redshift.service clean-trash.timer console.target weston.service homedir-gc@bbrepos.timer tablet.target xserver.target gpg-agent-update.service udev-ac.target clean-backups.service dbus.service homedir-gc@pics.timer adb.service basic.target termite@0.service mpd.service xmonad.service termite@3.service default.target kill-slice@apps.service dpms-on.service vncserver.target wm.slice clean-build.timer termite@6.service i3lock.service homedir-gc@docs.timer homedir-gc@repos.timer xrandr-mirror.service daemons.slice clean-tmp.timer xrandr-dual.service machine.slice worker.slice homedir-gc@repos.service dpms-off.service rtags.service daemons.target presentation.target synced.target gvfs-daemon.service pulseaudio.service homedir-gc@audio.timer yoga-tablet@all.service paths.target termite@4.service termite@1.service xorg.service -.slice homedir-gc@mail.service apps.slice desktop.target vifm-dropdown.service homedir-gc@lib.timer flag@synced.service emacs-proxy.service emacs.socket xkeymap.service vm-machine-qemu@arch.service udev-battery.target homedir-gc@vids.service termite@7.service lock.target mpd.socket homedir-gc@docs.service homedir-gc@mail.timer
-After=emacs-proxy.service emacs.socket weston.socket i3lock.service kill-slice@apps.service homedir-gc@repos.service tint2.service xrandr-single.service dpms-on.service homedir-gc@vids.timer plank.service clean-build.service xorg.service termite@2.service termite@10.service homedir-gc@music.service flag@synced.service termite@8.service clean-tmp.timer rtags.service rtags.socket adb-proxy.service dpms-off.service awesome.service homedir-gc@docs.service dbus.service homedir-gc@mail.service yoga-tablet@all.service homedir-gc@lib.timer xrdb.service adb.socket chat-away.service homedir-check@lib.timer journal-dropdown.service wm.slice termite@7.service homedir-gc@lib.service mpd.service termite@1.service weston.service gpg-agent-update.service flag.slice audio.slice gvfs-daemon.service openbox.service workers.slice termite@5.service xmonad.service homedir-check@lib.service vifm-dropdown.service homedir-gc@audio.timer homedir-gc@pics.timer homedir-gc@pics.service gpg-agent.service mpd.socket termite@3.service mpc-pause.service termite-dropdown.service jwatch.service homedir-gc@bbrepos.service gconf.service termite@6.service vm-disk-snapshot@virtdata.service clean-backups.timer stream-stop.service homedir-gc@mail.timer vm-disk-snapshot@arch.service clean-trash.service dbus.socket -.slice clean-backups.service homedir-gc@repos.timer chromium@0.service daemons.slice xrandr-mirror.service worker.slice homedir-gc@vids.service termite@0.service machine.slice vncserver.service emacs.service homedir-gc@docs.timer pulseaudio.socket emacs-save-quit.service xkeymap.service clean-trash.timer homedir-gc@audio.service homedir-gc@bbrepos.timer termite@9.service apps.slice compton.service clean-tmp.service clean-build.timer xrandr-dual.service homedir-gc@music.timer adb.service vm-machine-qemu@arch.service termite@4.service at-spi-bus.service redshift.service pulseaudio.service
+Wants=mpc-pause.service disconnect.target
+ConflictedBy=xrandr-mirror.service vncserver.service touchegg.service dconf.service jwatch.service xmonad.service audio.slice gpg-agent.service xrandr-single.service openbox.service default.target gvfsd-metadata.service daemons.slice xrandr-dual.service weston.service plank.service audio.target slices.target console.target yoga\x5cx2dtablet.slice gvfsd.service i3lock.service at-spi-bus.service tablet.target yoga-tablet@off.service presentation.target compton.service emacs.service tint2.service dpms-off.service apps.slice xorg.service gconf.service gpg-agent-update.service redshift.service setenv.service dbus.socket vncserver.target caribou.service desktop.target yoga-rotate@90.service tmux.service weston.socket yoga-tablet@all.service lock.target daemons.target timers.target wm.target dictd.service sockets.target pulseaudio.service yoga-rotate@0.service compositor.target dropdown.slice xrdb.service basic.target yoga\x5cx2drotate.slice dpms-timeout.service xkeymap.service paths.target caribou-antler.service gvfs-udisks2-volume-monitor.service dpms-on.service graphical.target pulseaudio.socket chat-away.service stream-stop.service xfconfd.service wm.slice awesome.service dbus.service laptop.target user.target mpc-pause.service xorg.socket
+After=xmonad.service wm.slice emacs.service i3lock.service daemons.slice dpms-on.service disconnect.target vncserver.service xrandr-single.service apps.slice audio.slice caribou-antler.service yoga\x5cx2dtablet.slice jwatch.service mpc-pause.service awesome.service redshift.service xkeymap.service dpms-off.service yoga-tablet@all.service dbus.service pulseaudio.socket plank.service xrandr-mirror.service gvfsd.service dropdown.slice stream-stop.service dictd.service tmux.service gpg-agent-update.service weston.service xorg.service tint2.service weston.socket xfconfd.service at-spi-bus.service xrdb.service pulseaudio.service caribou.service dbus.socket compton.service dpms-timeout.service yoga-rotate@0.service gvfsd-metadata.service chat-away.service yoga-rotate@90.service xorg.socket dconf.service gpg-agent.service yoga\x5cx2drotate.slice openbox.service yoga-tablet@off.service gvfs-udisks2-volume-monitor.service xrandr-dual.service gconf.service touchegg.service setenv.service
 Description=shutdown
 LoadState=loaded
 ActiveState=inactive
@@ -9,8 +10,7 @@
 FragmentPath=/home/abdo/.config/systemd/user/shutdown.target
 UnitFileState=static
 UnitFilePreset=enabled
-StateChangeTimestamp=Mon 2016-10-10 15:49:12 CEST
-StateChangeTimestampMonotonic=11846436421
+StateChangeTimestampMonotonic=0
 InactiveExitTimestampMonotonic=0
 ActiveEnterTimestampMonotonic=0
 ActiveExitTimestampMonotonic=0
@@ -37,3 +37,4 @@
 StartLimitIntervalSec=10000000
 StartLimitBurst=5
 StartLimitAction=none
```

**Comment by poettering** [comment]
Hmm, I think it might be an enumeration order issue here...

With the -.slice unit file around, does the problem actually go away?

**Comment by aroig** [comment]
Ok, that's weird. If I start my session on the laptop with `~/.config/systemd` empty the slice does not conflict with shutdown. So, some unrelated unit I have causes this.

Let me investigate further, it may be all my fault, although I still have no idea what happens, since I do not touch `-.slice` in my units...

**Comment by poettering** [comment]
I actually have the suspicion that this is a systemd bug. Basically my guess goes like this:

During initialization systemd will for each unit type "enumerate" the pre-existing registered objects that we want to track. For .device units it will for example enumerate all udev devices and create matching .device units for them in active state. For .mount units ti will enumerate everything in /proc/self/mountinfo and create matching .mount units in active state. Now, for slice units it actually doesn't do much, except for one thing: it synthesizes the root slice "-.slice". Now, during enumeration other units might get referenced and already loaded (for example, a device or mount pulling in some auxiliary service). My guess now is that this happens here and since all .service units implicitly depend on -.slice, what might happen is that -.slice gets referenced and loaded _before_ it gets enumerated. If it gets enumerated first, all is good because the enumeration code after all is able to synthesize -.slice fully. However, if it is loaded first, then it will already exist and get its deps by the time it is synthesized in the enumeratin code and the deps cannot be removed again...

That's why I am asking whether dropping in the -.slice unit for the user fixes the issue. Why would it? Well, the -.slice unit I added contains the exact same definitions that are syntesized during enumeration. Hence it wouldn't matter anymore whether the slice is enumerated first or loaded first, the same settings would result, and thus also dependencies.

Of course, if this really is the case, then the proper fix is more than just dropping in that -.slice file. We should also make sure that properly synthesize -.slice during load time too, if you follow what I mean...

**Comment by aroig** [comment]
Ah! I understand now.

And yup, dropping that `-.slice` into my config removes the `Conflicts=shutdown.target`.

**Comment by poettering** [comment]
ok, very good, then we should be able to fix this properly.

**Comment by poettering** [comment]
I prepped a more correct fix now in #4477.
